### PR TITLE
Fully qualify :grpc_python_plugin Bazel macro so it can be used from other repos.

### PR DESCRIPTION
**Pull request by g-easy**

### CODE REVIEW

**Comment by g-easy** [comment]
Without this change, I get errors like:
```
ERROR: [...]/googleapis/google/api/BUILD.bazel:537:1: no such target '//:grpc_python_plugin': target 'grpc_python_plugin' not declared in package '' defined by [...]/googleapis/BUILD.bazel and referenced by '//google/api:_annotations_py_proto_grpc_codegen'
```

cc @htuch @vam-google @bogdandrutu

File: bazel/python_rules.bzl [hunk 1]
```diff
@@ -163,7 +163,7 @@ def py_proto_library(
         _generate_py(
             name = codegen_grpc_target,
             deps = deps,
-            plugin = "//:grpc_python_plugin",
+            plugin = "@com_github_grpc_grpc//:grpc_python_plugin",
```
Comment (vam-google) [inline bazel/python_rules.bzl]: (Minor) I believe the idiomatic way of doing it is using [Label](https://docs.bazel.build/versions/master/skylark/lib/Label.html#Label) constructor directly:
```python
plugin = Label("//:grpc_python_binary"),
```
This allows resolving the label target correctly regardless of the actual name used for importing the repo.

Comment (g-easy) [reply bazel/python_rules.bzl]: Done. Thanks! I hadn't seen Label before.

**Comment by gnossen** [comment]
@g-easy Can you explain your use case? This was not a rule that we intended to make public. We're not opposed to doing so, but we don't want to unnecessarily tie ourselves to an API that hasn't gone through a thorough review. Once it's out in the wild, we're married to it.

We were actually hoping that this rule would just be an interim solution and we'd move to https://github.com/stackb/rules_proto once their python issues [are cleared up.](https://github.com/stackb/rules_proto/issues/86) (Speaking of which, they appear to have been resolved since I last looked.)

**Comment by g-easy** [comment]
I need to add py_proto_libraries to googleapis for envoy to consume. @vam-google asked me to use grpc's py_proto_library instead of the one from protobuf (which uses srcs="file.proto" instead of deps="proto_library_target").

**Comment by gnossen** [comment]
I'm willing to make the rule public with two caveats:
 1. The name should be changed to `grpc_py_proto_library` to indicate that this is *gRPC* Python generated code. This will be in line with the C++ rule. Since this was an internal rule before, this was implicit.
 2. We mark the rule experimental in its docstring.

You can accomplish the first item with
```bash
find src examples bazel -name 'BUILD' -o -name '*.bzl' -o -name BUILD.bazel | xargs sed -i 's/py_proto_library/grpc_py_proto_library/g'
```

**Comment by g-easy** [comment]
Could we please have separate rules for `py_proto_library` versus `py_grpc_library` to match the other languages? :)

**Comment by g-easy** [comment]
@vam-google - could you please elaborate why you wanted this rule instead of the one from protobuf?

Related issue: https://github.com/bazelbuild/bazel/issues/3935

**Comment by gnossen** [comment]
@g-easy This rule is [in line with C++](https://github.com/grpc/grpc/blob/master/bazel/cc_grpc_library.bzl#L6). Both rules by default generate both protobuf code and gRPC stubs. By setting the `proto_only` flag to true, you can produce just the protobuf code.

The missing piece is the `py_proto_library` rule referenced in the issue you linked. [C++ has one already](https://docs.bazel.build/versions/master/be/c-cpp.html#cc_proto_library). This is provided by the Blaze team internally. I would definitely like to mirror the internal structure as much as possible. And if that means gRPC team needs to take on that work, we can do that. But it will take a little time and coordination with Bazel team.

@dslomov Thoughts?

Let me elaborate on the problem a bit more deeply for context.

What we're seeing is an asymmetry between the rules provided internally and those available in open source. Internally, the ecosystem is as follows:
 - `proto_library` which takes in a `.proto` file and produces a language-agnostic descriptor file corresponding to that protobuf. This rule is provided by Blaze itself.
 - `py_proto_library` which takes the descriptors produced by `proto_library` and generates a Python module capable of serializing/deserializing that protobuf. This rule is also provided by Blaze itself.
 - `google3_py_grpc_library` which takes a descriptor file and protobuf python module and produces gRPC stubs. This rule is provided by the gRPC team.

So the `py_proto_library` rule in this repo is a combination of the functionality of the internal `py_proto_library` and `google3_py_grpc_library` rules. This naming was unfortunate, given the clash with the rule available internally.

**Comment by gnossen** [comment]
CC @nicolasnoble

**Comment by g-easy** [comment]
CC @kyessenov

**Comment by gnossen** [comment]
@g-easy From the response on https://github.com/bazelbuild/bazel/issues/3935, it doesn't sound like we should hold out hope for native Bazel support in the near term future. I'll work on a split-up of `py_proto_library` later today if that will get you unblocked.

**Comment by g-easy** [comment]
I think `py_proto_library` would be valuable for many projects. I have filed issue https://github.com/grpc/grpc/issues/19255 for this, and will close my PR since it doesn't look like it has anything in common with the long-term solution for this.

Thanks!

**Pull request closed without merging**